### PR TITLE
Add section titles to Manage Jenkins context menu

### PR DESCRIPTION
**Pull request by daniel-beck**
Alternative to #4583.

<details>
<summary>Screen shots</summary>

Screenshot 2020-03-16 at 13 27 58
Screenshot 2020-03-16 at 13 28 05

</details>

### Proposed changelog entries

* Add section headers to context menu of Manage Jenkins

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Comment by daniel-beck** [comment]
Reworking this to be less of a disgusting hack.

**Comment by daniel-beck** [comment]
Amended with larger font size (thanks @fqueiruga):

> 
Screenshot

File: core/src/main/resources/lib/layout/breadcrumbs.css [hunk 1]
```diff
@@ -126,3 +126,8 @@
 A.model-link.inside, #breadcrumbs A.inside {/* additional 'inside' class allows pre-allocation of the context menu space */
     padding-right: 16px;
 }
+
+#breadcrumb-menu .header {
```
Comment (fqueiruga) [inline core/src/main/resources/lib/layout/breadcrumbs.css]:
```suggestion
.breadcrumb-menu__category {
```

File: core/src/main/resources/lib/layout/breadcrumbs.js [hunk 2]
```diff
@@ -203,7 +203,11 @@ var breadcrumbs = (function() {
                 onComplete:function (x) {
                     var a = x.responseText.evalJSON().items;
                     function fillMenuItem(e) {
-                        e.text = makeMenuHtml(e.icon, e.displayName);
+                        if (e.header) {
+                            e.text = makeMenuHtml(e.icon, "<span class='header'>" + e.displayName + "</span>");
```
Comment (fqueiruga) [inline core/src/main/resources/lib/layout/breadcrumbs.js]:
```suggestion
                            e.text = makeMenuHtml(e.icon, "<span class='breadcrumb-menu__category'>" + e.displayName + "</span>");
```

**Review by fqueiruga — CHANGES_REQUESTED** [review]
Looks great! nice, smart solution. Just added a suggestion to rename the CSS class in order to make it more specific for this component.

**Comment by jtnord** [comment]
Have we considered how this will negativly effect scrolling...

e.g. it will make situations like the following much more common (and they are annoying)
image

**Review by fcojfernandez — APPROVED** [review]
Although I agree with James' comment, I think the benefits of having those titles are bigger than the effects on the scroll

**Comment by varyvol** [comment]
I like the change, I think it improves the visualization a lot. Happy to approve once you review @fqueiruga suggestions.

**Review by fqueiruga — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
I am fine with both alternatives, leaving the decision to UX experts.

P.S: We could even have both and select one based on the number of items (e.g. this PR when it is below 12 and #4583 when the number is above). But most likely it is an over-complication

**Comment by daniel-beck** [comment]
Personally I'd prefer this be merged, and the other being left as a workaround if feedback is overly negative for adding ~5 items.

**Comment by oleg-nenashev** [comment]
I plan to merge it tomorrow if no negative feedback

**Pull request merged**